### ISSUE TITLE
this OSXFUSE library version is incompatible with the OSXFUSE kernel extension

### ISSUE THREAD

**Issue by gnyman** [borgbackup/borg#1619]
Trying to mount a backup on OSX 10.11.6 fails with the above error.

Fuse for macOS version: 3.5.0 , installed with the package installer from https://osxfuse.github.io

./borg-macosx64 --version
borg-macosx64 1.1.0b1 

Tried the older (stable) version of borg also produces the same error.

I guess this is because borg is built against osxfuse-3.4.1 and the fix would be to build it against 3.5.0 which is listed as "stable" on the osxfuse release page https://github.com/osxfuse/osxfuse/releases

And indeed downloading and installing 3.4.1 from the osxfuse release page allows me to mount my backup again and browse it as expected.

Let me know if you need any more information, thank you for all your work with borg so far.

**Comment by ThomasWaldmann** [comment]
https://github.com/osxfuse/osxfuse/issues/313

**Comment by ThomasWaldmann** [comment]
As far as borg is concerned, the README.txt for the binary documents that it was made with 3.4.1.

I usually try to avoid .0 releaes and until recently, there was only 3.5.0. I did not notice it was tagged as stable release, but maybe that was just a mistake, see link to osxfuse ticket above. Now, 3.5.1 is out, so I might update that for next beta (depending on the answer from the osxfuse devs).

**Comment by ThomasWaldmann** [comment]
No bug, but keeping ticket open as osxfuse update reminder.

**Comment by gnyman** [comment]
Thanks for the quick reply, I did not even think about 3.5.0 being possibly marked as stable by mistake. But the fact that it is, means that the auto-update for osxfuse will update to it, so I guess it will be a popular version number and maybe a good idea to try to keep the borg binaries compatible with it. 

One would hope that there was a better way than having borg binaries break with each point release they do but I'm not familiar with how it works if it would be possible to make borg compatible with a wider range of versions.

**Comment by ThomasWaldmann** [comment]
The problem is not within borg, it just is that the osxfuse stuff (libraries) bundled with the borg binary needs to be compatible with the rest of osxfuse that is installed on your system (I suspect some kernel driver maybe?). Hopefully 3.5.0 will be compatible to 3.5.1.

**Comment by ThomasWaldmann** [comment]
upgraded osxfuse via #1649 to 3.5.1.

**Comment by bfleischer** [comment]
> The problem is not within borg, it just is that the osxfuse stuff (libraries) bundled with the borg binary needs to be compatible with the rest of osxfuse that is installed on your system (I suspect some kernel driver maybe?). Hopefully 3.5.0 will be compatible to 3.5.1.

You should not bundle any "osxfuse stuff" like libosxfuse with your file system. This will break your file system every time FUSE is upgraded. FUSE is a shared component and should be treated as such. Extracting some parts and shipping them with your app can cause issues.

**Comment by ThomasWaldmann** [comment]
I don't actively do that, but pyinstaller collects all dependencies and bundles them (on linux, that is everything except glibc related stuff, no idea what it does for OSX).

If there is an option to exclude your libs, I can try that.

**Comment by ThomasWaldmann** [comment]
```
MacBook-Pro:_MEIBzI7JI vagrant$ ls
Python                  _json.so                _socket.so              grp.so                  msgpack._packer.so
_bisect.so              _lzma.so                _ssl.so                 include                 msgpack._unpacker.so
_bz2.so                 _md5.so                 _struct.so              lib                     parser.so
_codecs_cn.so           _multibytecodec.so      array.so                lib2to3                 pyexpat.so
_codecs_hk.so           _multiprocessing.so     base_library.zip        libcrypto.1.0.0.dylib   readline.so
_codecs_iso2022.so      _opcode.so              binascii.so             liblz4.1.dylib          resource.so
_codecs_jp.so           _pickle.so              borg.chunker.so         liblzma.5.dylib         select.so
_codecs_kr.so           _posixsubprocess.so     borg.compress.so        libosxfuse.2.dylib      termios.so
_codecs_tw.so           _random.so              borg.crypto.so          libreadline.6.dylib     unicodedata.so
_ctypes.so              _scproxy.so             borg.hashindex.so       libssl.1.0.0.dylib      zlib.so
_datetime.so            _sha1.so                borg.platform.darwin.so llfuse.so
_hashlib.so             _sha256.so              borg.platform.posix.so  math.so
_heapq.so               _sha512.so              fcntl.so                mmap.so
```
So there is libosxfuse.2.dylib and llfuse.so (from python-llfuse).

So @bfleischer, you suggest we remove libosxfuse.2.dylib and the osxfuse stuff found on the system will still match the other binaries we bundle?

**Comment by ThomasWaldmann** [comment]
borg.exe.spec:
```
# -*- mode: python -*-

block_cipher = None


a = Analysis(['src/borg/__main__.py'],
             pathex=['/vagrant/borg/borg'],
             binaries=[],
             datas=[],
             hiddenimports=['borg.platform.posix'],
             hookspath=[],
             runtime_hooks=[],
             excludes=[],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher)

a.binaries = [b for b in a.binaries if 'libosxfuse' not in b[0]]

pyz = PYZ(a.pure, a.zipped_data,
             cipher=block_cipher)
exe = EXE(pyz,
          a.scripts,
          a.binaries,
          a.zipfiles,
          a.datas,
          name='borg.exe',
          debug=False,
          strip=False,
          upx=True,
          console=True )
```

**Comment by ThomasWaldmann** [comment]
Build for 1.1.0b2 done with modified spec file.

TODO: commit spec file

**Comment by ThomasWaldmann** [comment]
Fixed by PR #1673.
